frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx

```diff
@@ -12,58 +12,34 @@ const DevPortfolioForm: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const userInfo = useSelf()!;
 
-  const devPortfolios: DevPortfolio[] = [
-    {
-      name: 'test 1',
-      deadline: Date.parse('06 May 2022 00:00:00 GMT'),
-      earliestValidDate: Date.parse('01 May 2022 00:00:00 GMT'),
-      submissions: [],
-      uuid: 'xyz'
-    },
-    {
-      name: 'test 2',
-      deadline: Date.parse('20 Jan 2022 00:00:00 GMT'),
-      earliestValidDate: Date.parse('01 Jan 2022 00:00:00 GMT'),
-      submissions: [],
-      uuid: 'abc'
-    },
-    {
-      name: 'test abc',
-      deadline: Date.parse('30 Dec 2022 00:00:00 GMT'),
-      earliestValidDate: Date.parse('20 Nov 2022 00:00:00 GMT'),
-      submissions: [],
-      uuid: '123'
-    }
-  ];
-
-  // real!!!
   const [devPortfolio, setDevPortfolio] = useState<DevPortfolio | undefined>(undefined);
-  // const [devPortfolios, setDevPortfolios] = useState<DevPortfolio[]>([]);
+  const [devPortfolios, setDevPortfolios] = useState<DevPortfolio[]>([]);
   const [openPRs, setOpenPRs] = useState(['']);
   const [reviewPRs, setReviewedPRs] = useState(['']);
 
-  // useEffect(() => {
-  //   DevPortfolioAPI.getAllDevPortfolios().then((devPortfolios) => setDevPortfolios(devPortfolios));
-  // }, []);
+  useEffect(() => {
+    DevPortfolioAPI.getAllDevPortfolios().then((devPortfolios) => setDevPortfolios(devPortfolios));
+  }, []);
 
-  const requestDevPortfolio = (
+  const sendSubmissionRequest = (
     devPortfolioRequest: DevPortfolioSubmission,
     devPortfolio: DevPortfolio
   ) => {
-    devPortfolio?.submissions.push(devPortfolioRequest);
-    DevPortfolioAPI.requestDevPortfolio(devPortfolio).then((val) => {
-      if (val.error) {
-        Emitters.generalError.emit({
-          headerMsg: "Couldn't submit dev assignment!",
-          contentMsg: val.error
-        });
-      } else {
-        Emitters.generalSuccess.emit({
-          headerMsg: 'Dev Portfolio Assignment submitted!',
-          contentMsg: `The leads were notified of your submission and your submission will be graded soon!`
-        });
+    DevPortfolioAPI.makeDevPortfolioSubmission(devPortfolio.uuid, devPortfolioRequest).then(
+      (val) => {
+        if (val.error) {
+          Emitters.generalError.emit({
+            headerMsg: "Couldn't submit dev assignment!",
+            contentMsg: val.error
```

Comment:
@henry-li-06 Does that mean all of this should go too?

Reply:
I think it should be able to stay. I believe `responseMiddleware` just returns the response as long as it's not a 440 status code. And in the case of an error, the backend's response is always something like `{"error": <some-error-message>}`. 

You can see the error handling on the backend [here](https://github.com/cornell-dti/idol/blob/main/backend/src/api.ts#L104-L115). 

You can see [here](https://github.com/cornell-dti/idol/blob/main/frontend/src/components/Forms/UserProfile/UserProfile.tsx#L30-L42) that there's similar error handling on the frontend to what's in the dev portfolio form.